LoopKitUI/Extensions/Environment+Authenticate.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Environment+Authenticate.swift
+//  LoopKitUI
+//
+//  Created by Rick Pasetto on 8/3/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import LocalAuthentication
+import SwiftUI
+
+public typealias AuthenticationChallenge = (_ description: String, _ completion: @escaping (Result<Void, Error>) -> Void) -> Void
+public extension Result where Success == Void {
+    static var success: Result {
+        return Result.success(())
+    }
+}
+fileprivate struct UnknownError: Swift.Error { }
+
+private struct AuthenticationChallengeKey: EnvironmentKey {
+    static let defaultValue: AuthenticationChallenge = { authenticationChallengeDescription, completion in
+        let context = LAContext()
+        var error:NSError?
+        if context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
+            context.evaluatePolicy(.deviceOwnerAuthentication,
+                                   localizedReason: authenticationChallengeDescription,
+                                   reply: { (success, error) in
+                                    DispatchQueue.main.async {
+                                        assert(success || error != nil)
+                                        completion(success ? .success : .failure(error ?? UnknownError()))
+                                    }
+            })
+        } else {
+            assert(error != nil)
+            completion(.failure(error ?? UnknownError()))
```

Comment:
[nit/q] If we're asserting the error is `nil` here, then would it make sense to force unwrap it on the next line?

Reply:
I could see it either way; the assert won't be triggered in release builds, so if the framework had a bug where the assertion failed having a fallback isn't unreasonable.

Reply:
`assert` is compiled out in production code (https://developer.apple.com/documentation/swift/1541112-assert)
(note: `precondition` is not)
force-unwrapping would cause a crash in production code.  I felt it was unnecessary to do that, though it would not be unreasonable given that it would indicate Apple's APIs would violate their own contract, which is unlikely.  I could go either way.